common/types/ipv6.go

```diff
@@ -25,6 +25,12 @@ func (v6 IPv6) IP() net.IP {
 	return v6[:]
 }
 
+// To4 converts this address into an IPv4 address, for areas of the code which
+// reuse this structure for both IPv4 and IPv6 addresses.
+func (v6 IPv6) To4() net.IP {
+	return v6[:4]
```

Comment:
Seconded, we should maintain consistency with golang.

Reply:
This IPv6 address structure is intended to match the BPF representation, and should perhaps be instead placed somewhere like `pkg/bpf/types.go`. The implication of making such a change also involves updating the BPF code to understand the new format, and potentially impacts upgrade.

Question: Does golang treat `[]byte` and `[16]byte` as the same type? The main concern I see is if someone accidentally ends up inadvertently using this function using a different IP address type.

Reply:
Looking closer, the referred `IPv4(a, b, c, d)` will actually return a 16-byte array with the IPv4 address in the highest four bytes, but it's not consistent - look at https://golang.org/src/net/ip.go#L59 . Weirdly, if you then perform `String()` on an address created using `IPv4()`, it will be a 16-byte array so will be formatted as an IPv6 address. The `net.IP` type itself is just a `[]byte`, which means that in most cases an IPv4 address should be represented as `[4]byte`, not `[16]byte` with the address in the uppermost four bytes.